pkg/controller/backup_controller.go

```diff
@@ -281,6 +288,57 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	return backupLocation, validationErrors
 }
 
+// defaultAndValidateSnapshotLocations ensures:
+// - each location name in Spec VolumeSnapshotLocation exists as a location
+// - exactly 1 location per existing or default provider
+// - a given default provider's location name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
+func (c *backupController) defaultAndValidateSnapshotLocations(itm *api.Backup, defaultLocations map[provider]string) []string {
+	var errors []string
+	perProviderLocationName := make(map[provider]string)
+	var finalLocationNameList []string
+	for _, locationName := range itm.Spec.VolumeSnapshotLocations {
+		// validate each locationName exists as a VolumeSnapshotLocation
+		location, err := c.snapshotLocationLister.VolumeSnapshotLocations(itm.Namespace).Get(locationName)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("error getting volume snapshot location named %s: %v", locationName, err))
+			continue
+		}
+
+		// ensure we end up with exactly 1 locationName *per provider*
+		providerLocationName := perProviderLocationName[provider(location.Spec.Provider)]
+		fmt.Println("!!MMMMMM --- location.Spec.Provider ", location.Spec.Provider)
+		fmt.Println("!!!!!!i --- providerLocationName ", providerLocationName)
+		fmt.Println("!!YYYYY --- locationName ", locationName)
+		if providerLocationName != "" {
+			// if > 1 location name per provider as in ["aws-us-east-1" | "aws-us-west-1"] (same provider, multiple names)
+			if providerLocationName != locationName {
+				fmt.Println("!!! error!!!")
+				errors = append(errors, fmt.Sprintf("more than one VolumeSnapshotLocation specified for provider %s: %s", location.Spec.Provider, locationName))
+				continue
+			}
+		} else {
+			// no dup exists: add locationName to the final list
+			finalLocationNameList = append(finalLocationNameList, locationName)
+		}
+		// keep track of all valid existing locations, per provider
+		perProviderLocationName[provider(location.Spec.Provider)] = locationName
```

Comment:
I guess I'd probably put this in the `else` block since we only need to to it the first time we encounter a VSL for a provider. functionally the same but i think slightly clearer that way.